### PR TITLE
Fix quantization hang bugs.

### PR DESCRIPTION
**Pull request by wzzju**

### CODE REVIEW

**Review by panyx0718 — APPROVED** [review]

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py [hunk 1]
```diff
@@ -127,14 +121,17 @@ def apply(self, graph):
         """
         assert isinstance(graph,
                           IrGraph), 'graph must be the instance of IrGraph.'
-        self._need_initialized.clear()
+        #sequential_execution = core.get_pass('sequential_execution_pass')
+        #sequential_execution.apply(graph.graph)
```
Comment (wanghaoshuang) [inline python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:125]: Please remove unused comments.

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py [hunk 2]
```diff
@@ -742,9 +743,27 @@ def _insert_post_dequant_op(self, graph, op_node):
         graph.link_to(output_var_node, dequant_op_node)
         graph.link_to(scale_var_node, dequant_op_node)
         graph.link_to(dequant_op_node, dequant_var_node)
-        self._op_output_rename_map[output_var_node.name()] = dequant_var_node
+        self._op_output_rename_map[output_var_node.node] = dequant_var_node
         return dequant_var_node
 
+    def _init_var_node(self, var_node, value):
```
Comment (wanghaoshuang) [inline python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:749]: 和[line485](https://github.com/PaddlePaddle/Paddle/pull/16456/files#diff-896eadfd345b04fb29e4c215f8ce021cR485)的方法有什么区别么？

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py [hunk 3]
```diff
@@ -742,9 +743,27 @@ def _insert_post_dequant_op(self, graph, op_node):
...
+        assert isinstance(
+            value, np.ndarray), 'The type of value should be numpy array.'
+        assert self._scope is not None, \
+        'The scope cannot be set None when activation_quantize_type equals to range_abs_max.'
+        assert self._place is not None, \
+        'The place cannot be set None when activation_quantize_type equals to range_abs_max.'
+        tensor = self._scope.var(var_node.name()).get_tensor()
+        tensor.set(value, self._place)
+
+    def _to_node(self, nodes, node_name):
```
Comment (wanghaoshuang) [inline python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:759]: 这个方法命名有点不太直观。。。你是想name_to_node吧？或者find_node_by_name?

File: python/paddle/fluid/framework.py [hunk 4]
```diff
@@ -2312,6 +2341,34 @@ def safe_remove_nodes(self, remove_nodes):
         original_nodes = {n.node for n in remove_nodes}
         core.graph_safe_remove_nodes(self.graph, original_nodes)
 
+    def resolve_hazard(self):
+        def _to_node(nodes, node_name):
```
Comment (wanghaoshuang) [inline python/paddle/fluid/framework.py:2345]: 这个方法又实现了一遍？可以复用么？

File: python/paddle/fluid/framework.py [hunk 5]
```diff
@@ -2312,6 +2341,34 @@ def safe_remove_nodes(self, remove_nodes):
...
+            for n in nodes:
+                if n.name() == node_name:
+                    target_node = n
+            assert target_node is not None, "Cannot find the target node in the giving set."
+            return target_node
+
+        ordered_nodes = core.topology_sort(self.graph)
+        var_nodes = dict()
+        for node in ordered_nodes:
+            if node.is_op() and node.op() is not None:
```
Comment (wanghaoshuang) [inline python/paddle/fluid/framework.py:2356]: 什么情况下，node.op() is None?

File: python/paddle/fluid/framework.py (hunk 4 above)
Comment (wzzju) [reply python/paddle/fluid/framework.py:2345]: 不在同一个文件里面，`_to_node`只是一个辅助函数

File: python/paddle/fluid/framework.py (hunk 5 above)
Comment (wzzju) [reply python/paddle/fluid/framework.py:2356]: ctr_node或者empty node

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py (hunk 1 above)
Comment (wzzju) [reply python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:125]: 可能需要，正在测试。如果不需要后面会去掉。

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py (hunk 2 above)
Comment (wzzju) [reply python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:749]: 不在一个类里面，但是都需要这个功能。

File: python/paddle/fluid/contrib/slim/quantization/quantization_pass.py (hunk 3 above)
Comment (wzzju) [reply python/paddle/fluid/contrib/slim/quantization/quantization_pass.py:759]: 这个是个辅助函数，内部使用的，感觉这样简洁一点？

**Comment by luotao1** [comment]
请问这是会影响windows hang么？
```
[03:57:12]	457/464 Test #463: test_quantization_pass ..........................***Failed   15.05 sec
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	.
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	
[03:57:12]	This application has requested the Runtime to terminate it in an unusual way.
[03:57:12]	Please contact the application's support team for more information.
[03:57:12]	W0327 11:57:01.409394 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
[03:57:12]	W0327 11:57:04.076364 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
[03:57:12]	W0327 11:57:06.304749 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
[03:57:12]	W0327 11:57:08.452008 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
[03:57:12]	W0327 11:57:10.598065 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
```
http://ci.paddlepaddle.org/viewLog.html?tab=buildLog&buildTypeId=PaddleWindows_PrWindowsCi&buildId=76062
Description能不能写的详细点？

**Comment by wzzju** [comment]
> 请问这是会影响windows hang么？
> 
> ```
> [03:57:12]	457/464 Test #463: test_quantization_pass ..........................***Failed   15.05 sec
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	.
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	
> [03:57:12]	This application has requested the Runtime to terminate it in an unusual way.
> [03:57:12]	Please contact the application's support team for more information.
> [03:57:12]	W0327 11:57:01.409394 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
> [03:57:12]	W0327 11:57:04.076364 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
> [03:57:12]	W0327 11:57:06.304749 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
> [03:57:12]	W0327 11:57:08.452008 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
> [03:57:12]	W0327 11:57:10.598065 27508 graph.h:204] WARN: After a series of passes, the current graph can be quite different from OriginProgram. So, please avoid using the `OriginProgram()` method!
> ```
> http://ci.paddlepaddle.org/viewLog.html?tab=buildLog&buildTypeId=PaddleWindows_PrWindowsCi&buildId=76062
> Description能不能写的详细点？

windows那个问题，暂时无解，看看这个pr能否修复。而且windows那个也不是hang的问题，我这个pr修复的是之前和里面的reviewer讨论过的问题。

**Pull request merged**